### PR TITLE
:art: fix issue #22, and fix test cases

### PR DESCRIPTION
**Pull request by ezrayst**

### CODE REVIEW

File: include/gmsh.h [hunk 1]
```diff
@@ -34,9 +34,6 @@ class GMSH : public Mesh<Tdim, Tvertices> {
   //! Compute material points from element coordinate map
   void compute_material_points();
 
-  //! Compute stresses
```
Comment (kks32) [inline include/gmsh.h]: Why is compute stress function removed?

File: include/gmsh.tcc [hunk 2]
```diff
@@ -193,6 +193,13 @@ void GMSH<Tdim, Tvertices>::compute_material_points() {
 
   const unsigned firstelementcoord = elementcoordinates_.begin()->first;
   const unsigned lastelementcoord = elementcoordinates_.rbegin()->first;
+  // Material density
```
Comment (kks32) [inline include/gmsh.tcc]: Please separate create material points from the stress computation.

File: include/io.h [hunk 3]
```diff
@@ -40,7 +40,7 @@ class IO {
 
   //! Write stresses
   void write_stresses(
-      const std::vector<std::array<double, Tdim * 2>>& stresses);
```
Comment (kks32) [inline include/io.h]: Write stresses should only need a vector of doubles, not a shared pointer.

File: include/mesh.h [hunk 4]
```diff
@@ -24,9 +24,6 @@ class Mesh {
   //! Compute material point location
   virtual void compute_material_points() = 0;
 
-  //! Compute initial stresses for material points
```
Comment (kks32) [inline include/mesh.h]: This is required. Compute stress should be a separate function.

File: include/point.h [hunk 5]
```diff
@@ -11,8 +11,12 @@ class Point {
   //! Constructor with id and coordinates
   //! \param[in] id index of the vertex
   //! \param[in] coord Coordinates of the point
-  Point(unsigned id, const std::array<double, Tdim>& coord) : id_{id} {
```
Comment (kks32) [inline include/point.h]: Constructor with just ID and coordinates.

File: include/point.h [hunk 6]
```diff
@@ -11,8 +11,12 @@ class Point {
...
+  //! \param[in] stress Voigt stress of the point
+  Point(unsigned id, const std::array<double, Tdim>& coord,
+        const std::array<double, Tdim * 2>& stress) {
+    id_ = id;
     coordinates_ = coord;
+    stress_ = stress;
```
Comment (kks32) [inline include/point.h]: Stress should be initialised later.

File: tests/point_test.cc [hunk 7]
```diff
@@ -16,39 +16,69 @@ TEST_CASE("Point base is checked in 2D", "[Point][2D]") {
   SECTION("Coordinates are checked for zero values") {
     const unsigned index = 0;
     const std::array<double, DIM> coordinate{0, 0};
-    auto node = std::unique_ptr<Point<DIM>>(new Point<DIM>(index, coordinate));
+    const std::array<double, DIM * 2> stress{0, 0, 0, 0};
```
Comment (kks32) [inline tests/point_test.cc]: This should be computed to verify, not assign and verify.

**Review by kks32 — CHANGES_REQUESTED** [review]

File: include/io.h (hunk 3 above)
Comment (ezrayst) [reply include/io.h]: In the next push, I will still have `materialpoints` as the input. Please let me know how to change them

Comment (ezrayst) [reply include/io.h]: Also, you don't want `stress` in `mesh` anymore?

Comment (kks32) [reply include/io.h]: No, I don't want `stress` in `mesh`, stress is a property of `material_point`

File: include/io.tcc [hunk 8]
```diff
@@ -34,7 +34,7 @@ void IO<Tdim>::write_material_points(
 //! /tparam Tdim dimension
 template <unsigned Tdim>
 void IO<Tdim>::write_stresses(
-    const std::vector<std::array<double, Tdim * 2>>& stresses) {
+    const std::vector<std::shared_ptr<Point<Tdim>>>& materialpoints) {
```
Comment (kks32) [inline include/io.tcc]: Iterate over `materialpoints` create a vector of stresses and pass it to this function.

File: include/point.h (hunk 5 above)
Comment (kks32) [reply include/point.h]: Revert to `Point(unsigned id, const std::array<double, Tdim>& coord) : id_{id} {`

File: include/point.h [hunk 9]
```diff
@@ -21,11 +22,21 @@ class Point {
   //! Return coordinates of the point
   std::array<double, Tdim> coordinates() const { return coordinates_; }
 
+  //! Return stresses of the point
+  std::array<double, Tdim * 2> stress() const { return stress_; }
+
+  //! Set stress
```
Comment (kks32) [inline include/point.h]: `//! Assign stresses`.

File: include/point.h [hunk 10]
```diff
@@ -21,11 +22,21 @@ class Point {
...
+  void set_stress(const std::array<double, Tdim * 2>& stress) {
```
Comment (kks32) [inline include/point.h]: Overload this function call it `stress`

**Review by kks32 — CHANGES_REQUESTED** [review]

File: include/io.tcc (hunk 8 above)
Comment (ezrayst) [reply include/io.tcc]: do you want to make a new function in `io` to iterate?

**Review by kks32 — APPROVED** [review]

**Pull request merged**